mm/mm_heap/mm_malloc.c

```diff
@@ -87,6 +87,14 @@ void mm_dump_handler(FAR struct tcb_s *tcb, FAR void *arg)
 }
 #endif
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* all zero allocated pointers to a global unique ptr */
+
+uint8_t g_zero_alloc;
```

Comment:
```suggestion
const uintptr_t g_zero_alloc;
```

1. add const to catch the caller write to it
2. change uintptr_t to get the correct alignment

Reply:
const address can not implicit cast to void * pointer, need explicit cast, but also lose read-only protection
> 1. add const to catch the caller write to it   

OK
> 2. change uintptr_t to get the correct alignment

Reply:
> const address can not implicit cast to void * pointer, need explicit cast, but also lose read-only protection
> 

Yes, we need cast here, but the benefit is that CPU can catch the wrong usage since the constant data normally locate in flash when XIP, or forbid to write by MMU/MPU.

Reply:
removed global variable.